merchant-sdk/sdk/src/main/java/net/gini/android/merchant/sdk/integratedFlow/MerchantFragment.kt

```diff
@@ -0,0 +1,270 @@
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+import kotlinx.parcelize.Parcelize
+import net.gini.android.merchant.sdk.GiniMerchant
+import net.gini.android.merchant.sdk.R
+import net.gini.android.merchant.sdk.bankselection.BankSelectionBottomSheet
+import net.gini.android.merchant.sdk.databinding.GmsFragmentMerchantBinding
+import net.gini.android.merchant.sdk.moreinformation.MoreInformationFragment
+import net.gini.android.merchant.sdk.paymentComponentBottomSheet.PaymentComponentBottomSheet
+import net.gini.android.merchant.sdk.paymentcomponent.PaymentComponent
+import net.gini.android.merchant.sdk.paymentcomponent.SelectedPaymentProviderAppState
+import net.gini.android.merchant.sdk.review.ReviewConfiguration
+import net.gini.android.merchant.sdk.review.ReviewFragment
+import net.gini.android.merchant.sdk.util.BackListener
+import net.gini.android.merchant.sdk.util.DisplayedScreen
+import net.gini.android.merchant.sdk.util.autoCleared
+import net.gini.android.merchant.sdk.util.extensions.add
+import net.gini.android.merchant.sdk.util.getLayoutInflaterWithGiniMerchantTheme
+import net.gini.android.merchant.sdk.util.wrappedWithGiniMerchantTheme
+import org.jetbrains.annotations.VisibleForTesting
+
+/**
+ * Configuration for the integrated payment flow
+ */
+@Parcelize
+data class MerchantFlowConfiguration(
+    /**
+     * If set to `true`, the [ReviewFragment] will be shown before initiating payment.
+     * If set to `false`, the payment request process will be executed under the hood before redirecting to the selected payment provider
+     *
+     * Default value is `false`
+     */
+    val shouldShowReviewFragment: Boolean = false,
+
+    /**
+     * If set to `true`, the errors will be handled internally and snackbars will be shown for errors.
+     * If set to `false`, errors will be ignored by the [MerchantFragment] and the [ReviewFragment]. In this case the flows exposed by [GiniMerchant] should be observed for errors.
+     *
+     * Default value is `true`.
+     */
+    val shouldHandleErrorsInternally: Boolean = true,
+
+    /**
+     * If set to `true`, the [Amount] field of shown as part of the [ReviewFragment] will be editable.
+     * If set to `false`, the [Amount] field will be read-only.
+     *
+     * Default value is `false`
+     */
+    val isAmountFieldEditable: Boolean = false
+): Parcelable
+
+/**
+ * The [MerchantFragment] provides a container for all screens that should be displayed for the user
+ * during the payment process (eg. [PaymentComponentBottomSheet], [BankSelectionBottomSheet], [ReviewFragment]).
+ *
+ * It handles the display logic for all screens. A new instance can be created using the [PaymentComponent.getContainerFragment] method.
+ */
+class MerchantFragment private constructor(
```

Comment:
@danicretu I am sorry for making you change the name again, but [naming is hard](https://martinfowler.com/bliki/TwoHardThings.html)... I think we should not name it MerchantFragment. Reading this makes me think of some UI related to a merchant (like viewing some details about the merchant). Knowing that it actually is about performing a payment makes me think that this is name is not a good fit at all 😞 

What do you think about `PaymentFlowFragment`? It reflects that it is about a payment and that it manages the whole flow.